components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m

```diff
@@ -0,0 +1,667 @@
+  CGFloat contentHeaderHeight = self.contentHeaderHeight;
+  CGFloat containerHeight = self.presentingViewBounds.size.height;
+
+  contentHeight += addedContentHeight;
+  _addedContentHeight = addedContentHeight;
+
+  CGFloat totalHeight = contentHeight + contentHeaderHeight;
+  CGFloat contentHeightThresholdForScrollability =
+      containerHeight * kMDCBottomDrawerInitialDrawerHeightFactor + contentHeaderHeight;
+  BOOL contentScrollsToReveal = totalHeight > contentHeightThresholdForScrollability;
+
+  if (_contentHeaderTopInset == NSNotFound) {
+    // The content header top inset is only set once.
+    if (contentScrollsToReveal) {
+      _contentHeaderTopInset = containerHeight * (1.f - kMDCBottomDrawerInitialDrawerHeightFactor);
+    } else {
+      _contentHeaderTopInset = containerHeight - totalHeight;
+    }
+  }
+
+  CGFloat scrollingDistance = _contentHeaderTopInset + contentHeaderHeight + contentHeight;
+  _contentHeightSurplus = scrollingDistance - containerHeight;
+
+  if (addedContentHeight < FLT_EPSILON && (_contentHeaderTopInset > _contentHeightSurplus) &&
+      (_contentHeaderTopInset - _contentHeightSurplus < self.addedContentHeightThreshold)) {
+    CGFloat addedContentheight = _contentHeaderTopInset - _contentHeightSurplus;
+    [self cacheLayoutCalculationsWithAddedContentHeight:addedContentheight];
+  }
+}
+
+- (CGFloat)transitionPercentageForContentOffset:(CGPoint)contentOffset
+                                         offset:(CGFloat)offset
+                                       distance:(CGFloat)distance {
+  return 1.f - MAX(0.f, MIN(1.f, (self.transitionCompleteContentOffset - contentOffset.y - offset) /
+                                     distance));
+}
+
+- (CGFloat)midAnimationScrollToPositionForOffset:(CGPoint)targetContentOffset {
+  if (!self.contentScrollsToReveal) {
+    return NSNotFound;
+  }
+
+  CGFloat headerAnimationDistance = self.headerAnimationDistance;
+  CGFloat headerTransitionToTop =
+      [self transitionPercentageForContentOffset:targetContentOffset
+                                          offset:0
+                                        distance:headerAnimationDistance];
+  if (headerTransitionToTop >= FLT_EPSILON && headerTransitionToTop < 1.f) {
+    CGFloat contentHeaderFullyCoversTopHeaderContentOffset = self.transitionCompleteContentOffset;
+    CGFloat contentHeaderReachesTopHeaderContentOffset =
+        contentHeaderFullyCoversTopHeaderContentOffset - headerAnimationDistance;
+    return self.scrollViewIsDraggedToBottom ? contentHeaderReachesTopHeaderContentOffset
+                                            : contentHeaderFullyCoversTopHeaderContentOffset;
+  }
+
+  return NSNotFound;
+}
+
+@end
+
+#pragma mark - MDCBottomDrawerContainerViewController + Layout Values
+
+@implementation MDCBottomDrawerContainerViewController (LayoutValues)
+
+- (CGRect)presentingViewBounds {
+  return self.originalPresentingViewController.view.bounds;
```

Comment:
As this is used in many places for geometry calculations, consider returning CGRectStandardize(self.originalPresentingViewController.view.bounds), but if you do so please appledoc this function and make it clear that this is standardized.
Ideally we would name it presentingViewStandardizedBounds, but I understand how that would defeat the purpose.

Reply:
done and documented